pkgs/by-name/gn/gnucash/package.nix

```diff
@@ -94,6 +103,20 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  postPatch = ''
+    patch -p0 <<END_PATCH
```

Comment:
Please commit the patch as a file to make working with it easier

Reply:
This patch requires templating some directories which are in scope here. How would that work if the patch file is separate? :thinking:

Reply:
We can use subtitute to replace those

Reply:
Thanks for the hint! The Nix documentation has led me to `replaceVars`. I now have the following added to `patches`:
```nix
    (replaceVars ./0005-python-env.patch {
      gnc_dbd_dir = "${libdbiDrivers}/lib/dbd";
      gsettings_schema_dir = glib.makeSchemaPath "${placeholder "out"}" "gnucash-${version}";
    })
```
This does replace the respective placeholders in the patch file, but `placeholder "out"` resolves to the path of the patch itself and not the gnucash derivation. I am not sure what I should alter to fix this :grimacing:

Reply:
Unsure if there are other good ways, I solved it [here](https://github.com/nevivurn/nixpkgs/blob/123ae57041b56b40685b7c752df51714f25fbf38/pkgs/servers/web-apps/kavita/default.nix#L44-L49) by creating the patch with [`@vars@`](https://github.com/nevivurn/nixpkgs/blob/123ae57041b56b40685b7c752df51714f25fbf38/pkgs/servers/web-apps/kavita/change-webroot.diff#L23) and and substituting them again in postPatch.

Reply:
Thanks. I went the same route here with a distinct patch and subsitution